packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -393,9 +392,11 @@ class ThemeData {
     IconThemeData accentIconTheme,
     TargetPlatform platform,
   }) {
+    // ThemeData objects are constructed by ThemeData.raw() and do not have
+    // any null-valued properties. All of the constructor parameters below
+    // will also be non-null and its why a primarySwatch parameter isn't needed.
```

Comment:
I do not know why it was written that way. 

I've replaced the comment and the ordinary ThemeData constructor with ThemeData.raw.